integration/deploy_test.go

```diff
@@ -98,6 +98,15 @@ func TestDeployTail(t *testing.T) {
 	WaitForLogs(t, out, "Hello world!")
 }
 
+func TestDeployTailDefaultNamespace(t *testing.T) {
+	MarkIntegrationTest(t, CanRunWithoutGcp)
+
+	// `--default-repo=` is used to cancel the default repo that is set by default.
+	out := skaffold.Deploy("--tail", "--images", "busybox:latest", "--default-repo=").InDir("testdata/deploy-hello-tail").RunLive(t)
+
+	WaitForLogs(t, out, "Hello world!")
+}
+
```

Comment:
The reason we moved to random namespace was that tests running in parallel don't interfere.  We used to see a bunch of flakes before.
How are we solving that here?

Reply:
the `out` stream returned by `RunLive()` is the same stream you would see doing a normal `skaffold run --tail`, which means that the output there is already filtered by the `runID` injected by skaffold when deploying. this implies that the `WaitForLogs()` function will only be checking the logs for the current test run, which is what we want.

we would see problems with tests stepping on each other when waiting for the pods/deployments to be stable, but I removed those checks from these tests since we only care about the logging behavior.

Reply:
sounds good!